src/App.tsx

```diff
@@ -1,39 +1,84 @@
-import { useState } from 'react';
-import { Routes, Route } from 'react-router-dom';
 import Box from '@mui/material/Box';
 import Button from '@mui/material/Button';
 import { utils } from 'ethers';
+import { useState } from 'react';
+import { Route, Routes } from 'react-router-dom';
 import errorIcon from './assets/icons/error.svg';
+import loadingIcon from './assets/icons/loading.svg';
+import successIcon from './assets/icons/success.svg';
 import { AppDialog } from './components/AppDialog/AppDialog.component';
+import { DateSelector } from './components/DateSelector/DateSelector.component';
 import { Header } from './components/Header/Header.component';
 import { InputWithButtonPillGroup } from './components/InputPillGroup/InputWithButtonPillGroup.component';
+import { MintingInfo } from './components/MintingInfo/MintingInfo.component';
+import { MintingProcessInfo } from './components/MintingInfo/MintingInfo.types';
 import { Urls } from './constants';
 import { DashboardContainer } from './pages/Dashboard/Dashboard.container';
 import { ProposalDetailsContainer } from './pages/ProposalDetails/ProposalDetails.container';
 import { ProposalsListContainer } from './pages/ProposalsList/ProposalsList.container';
 import { StakingContainer } from './pages/Staking/Staking.container';
-import { DateSelector } from './components/DateSelector/DateSelector.component';
 
 function App() {
   const [selectedDate, setSelectedDate] = useState<number>();
-  const [isOpenDialog, setIsOpenDialog] = useState(false);
+  const [openDialogType, setOpenDialogType] = useState<string | null>(null);
```

Comment:
let's use union as a type here. Also (that's completely personal opinion, and discussion purpose) I don't really like using `null` and `undefined` in the same apa. As React is evaluating empty state to `undefined`, I would argue for not using `null` at all.
```suggestion
  const [openDialogType, setOpenDialogType] = useState<'success' | 'error'>();
```